linode/manager
# feat: [M3-7373] - Sold Out Chips

## Description 📝
This PR implements sold out chips to our `PlanSelection` components (both Linode and Kubernetes). It helps our users with picking a `GPU` or `Premium CPU` plan by showing them plans that are actually available to them instead of trying to select unavailable plans that would lead to an API error upon submission.

It utilizes the new regions/availability endpoint which only returns availability for `GPU` or `Premium CPU` plans (it may be expanded later on). In the background, the endpoint gets refreshed at 15m via a cron job checking available resources in our DCs.

⚠️: While the endpoint is already live in production, we're waiting for an API update that will convert the `plan` value to our regular plan Ids. This is why this PR is to be tested with mock data.

⚠️: The feature flag is currently turned on for testing, but will be off in production till the fix above is completed. The flag controls the enabling of the API query so when `off` and no data is returned no plans will be shown as sold out. 

## Changes  🔄
- Implement logic to add the sold out chip to `GPU` or `Premium CPU` plans
- Add test coverage for new feature
- Modify the `useRegionsAvailabilitiesQuery` to be abled to be disabled
- Add ability to pass JSX element to `subheadings` mobile selection cards
- Get rid of obsolete `soldOutSingapore` & `soldOutTokyo` flags
- Replace instances of `selectedRegionID` with `selectedRegionId`


## Preview 📷

### Linode Creation Flow

| Mobile  | Desktop   |
| ------- | ------- |
| ![Screenshot 2023-12-21 at 11 58 57](https://github.com/linode/manager/assets/130582365/c7f8323e-e852-4b43-947e-44969904e008) | ![Screenshot 2023-12-21 at 11 59 04](https://github.com/linode/manager/assets/130582365/b19f5825-398a-48d9-953f-f7469efe1486) |

### Kubernetes Creation Flow
| Mobile  | Desktop   |
| ------- | ------- |
| ![Screenshot 2023-12-21 at 12 00 07](https://github.com/linode/manager/assets/130582365/92529529-8c2a-4ec4-bced-13faf08ed932) | ![Screenshot 2023-12-21 at 12 00 14](https://github.com/linode/manager/assets/130582365/984bff1b-770b-42fd-975b-71401b722fb1) |

## How to test 🧪

### Prerequisites
- pull and run code locally
- turn on MSW

### Verification steps 
for both the Linode and Kubernetes creation flows: 
- select either Newark, NJ (us-east) or Dallas, TX (us-central)
- confirm sold out chips in either the Dedicated CPU or Shared CPU tabs (those plans are not representative of what is returned by the API but what was mocked for testing purposes)
- Confirm mobile and desktop styling
- Confirm light and dark mode (ignore known visual defect for disabled radios which will be released at the same time)
- Confirm chip doesn't show if panel is disabled (you can do that by making [this prop](https://github.com/linode/manager/blob/develop/packages/manager/src/features/components/PlansPanel/PlansPanel.tsx#L85) `true`)

## As an Author I have considered 🤔

*Check all that apply*
